src/services/datasources/hooks/validateData.js

```diff
@@ -0,0 +1,19 @@
+const { BadRequest } = require('@feathersjs/errors');
+
+/**
+ * checks for the fields required to create or patch a datasource, and filters any additional fields.
+ */
+module.exports = (context) => {
+	if (context.method === 'create' && !(context.data.config)) {
+		throw new BadRequest('this requires a config object.');
+	}
+	if (context.data.config && !context.data.config.type) {
+		throw new BadRequest('config should contain a type');
```

Comment:
Why not use json schema here as well?

Reply:
because it was written before I brought in JSON schema... done :)